### ISSUE TITLE
Inefficient allocations in worker pool

### ISSUE THREAD

**Issue by adamglt** [neo4j/neo4j-go-driver#47]
**Driver version**: Go driver 1.7.2

In the new worker pool - there's this function:
```
func workerEntryPoint(pool *workerPool) {
	for {
		select {
		case work := <-pool.workQueue:
			work(pool.stopper)
		case <-pool.stopper:
			return
		case <-time.After(pool.keepAlive):
			return
		}
	}
}
```

We've seen some GC pressure caused by this: `time.After` creates a new timer for every call, so every loop iteration creates a timer.  
In our integration tests this actually brings `time.NewTimer` to the top of memory allocations:
```
(pprof) top
Showing nodes accounting for 19004.54kB, 84.13% of 22588.77kB total
Showing top 10 nodes out of 172
      flat  flat%   sum%        cum   cum%
 4608.45kB 20.40% 20.40%  4608.45kB 20.40%  github.com/<omitted>/vendor/github.com/neo4j-drivers/gobolt.(*workerConnection).queueJob
 4608.38kB 20.40% 40.80%  4608.38kB 20.40%  time.NewTimer
```

I think we can fix this relatively easily, but I haven't dived too deep into the code. Happy to hear your thoughts.

The change:
```
func workerEntryPoint(pool *workerPool) {
	t := time.NewTimer(pool.keepAlive)
	for {
		select {
		case work := <-pool.workQueue:
			work(pool.stopper)
			if !t.Stop() {
				<-t.C
			}
			t.Reset(pool.keepAlive)
		case <-pool.stopper:
			return
		case <-t.C:
			return
		}
	}
}
```

After change: (7MB drop, NewTimer doesn't even sample)
```
(pprof) top
Showing nodes accounting for 10760.55kB, 75.01% of 14344.66kB total
Showing top 10 nodes out of 145
```

**Comment by ali-ince** [comment]
Hi @adamglt,

Thanks for reporting this issue. Your proposed change looks good to me as well.

I'll be happy to merge it if you could create a PR at https://github.com/neo4j-drivers/gobolt for `1.7-dev` branch.

Ali

**Comment by adamglt** [comment]
Sure - created.

**Comment by ali-ince** [comment]
Thanks @adamglt, the PR is now merged, will be part of 1.7.3 release in the upcoming weeks.
